src/Rings/MPolyMap/flattenings.jl

```diff
@@ -263,6 +411,30 @@ function map_from_coefficient_ring_to_flattening(phi::RingFlattening)
   return phi.base_ring_to_flat
 end
 
+### flat counterparts of algebraic objects defined over the domain
```

Comment:
Here I explain the philosophy of caching the `flat_counterpart`s. One issue might be that everything with cached flat counterpart must be "mathematically immutable". I think that's more or less safe in Oscar, but one probably has to be careful with things like in-place operations on matrices. Does anyone see problems here?